app/src/main/java/ch/sdp/vibester/activity/PartyRoomActivity.kt

```diff
@@ -1,49 +1,143 @@
 package ch.sdp.vibester.activity
 
 import android.content.ContentValues
+import android.content.Intent
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
 import android.util.Log
+import android.widget.Button
 import android.widget.TextView
+import androidx.constraintlayout.widget.ConstraintLayout
 import ch.sdp.vibester.R
+import ch.sdp.vibester.api.LastfmApiInterface
+import ch.sdp.vibester.api.LastfmMethod
+import ch.sdp.vibester.api.LastfmUri
+import ch.sdp.vibester.database.AppPreferences
 import ch.sdp.vibester.database.DataGetter
 import ch.sdp.vibester.database.Database
+import ch.sdp.vibester.helper.GameManager
 import ch.sdp.vibester.helper.PartyRoom
 import ch.sdp.vibester.user.User
 import com.google.firebase.database.DataSnapshot
 import com.google.firebase.database.DatabaseError
 import com.google.firebase.database.ValueEventListener
+import com.google.gson.Gson
 import dagger.hilt.android.AndroidEntryPoint
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
 import javax.inject.Inject
+import kotlin.properties.Delegates
 
 @AndroidEntryPoint
 class PartyRoomActivity : AppCompatActivity() {
     @Inject
     lateinit var dataGetter: DataGetter
 
+    lateinit var gameManager: GameManager
+
+    lateinit var roomID: String
+
+
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_party_room)
 
-        val roomName = intent.getStringExtra("roomName").toString()
         val createPartyRoom = intent.getBooleanExtra("createRoom", false)
         if(createPartyRoom) {
-            createRoom(roomName)
+            createRoom()
+            setGameManager()
+            dataGetter.updateRoomField(roomID, "gameStarted", false)
         }
         else {
-            fetchData(roomName)
+            this.roomID = intent.getStringExtra("roomID").toString()
+            fetchData(roomID)
         }
+
+        val startGame = findViewById<Button>(R.id.startGame)
+
+        startGame.setOnClickListener {
+//            dataGetter.updateStartGame(roomID, true)
```

Comment:
Is this necessary, or will it be in the future?

Reply:
Good spot, will delete